Inventory.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using InventoryManagementSystem.ProductInfo;
+
+namespace InventoryManagementSystem.InventoryInfo
+{
+    public class Inventory
+    {
+        private readonly int _Id;
+        public static int InventoryIdCounter { get; private set; }
+        public string InventoryName { get; set; }
+        private List<Product> ProductList;
+
+        public Inventory() : this("DefaultInventory")
+        {
+
+        }
+
+        public Inventory(string inventoryName)
+        {
+            InventoryName = inventoryName;
+            ProductList = new List<Product>();
+            _Id = InventoryIdCounter++;
+        }
+
+        public override string ToString() => $"Id {_Id} : Name = {InventoryName}";
+
+        public void AddProduct()
+        {
+            Console.Write("Product name : ");
+            var prodcutName = Console.ReadLine();
+            Console.Write("Product price : ");
+            var prodcutPrice = Console.ReadLine();
+            Console.Write("Product quantity : ");
+            var prodcutQuantity = Console.ReadLine();
+
+            var product = new Product()
+            {
+                ProductName = prodcutName,
+                ProductPrice = Double.Parse(prodcutPrice),
+                ProductQuantity = int.Parse(prodcutQuantity)
+            };
+            ProductList.Add(product);
+        }
+
+        public void ViewProdcuts()
+        {
+            foreach(var product in ProductList)
+            {
+                product.Log();
+            }
+        }
+
+        public bool SearchForProduct(string productName)
+        {
+            foreach (var product in ProductList)
+            {
+                if(product.ProductName == productName)
+                {
+                    product.Log();
+                    return true;
+                }
+            }
+            Console.WriteLine("Product was not found");
+            return false;
+        }
+
+        public void EditProduct (string productName)
+        {
+            foreach (var product in ProductList)
+            {
+                if (product.ProductName == productName)
+                {
+                    Console.Write("Product new name : ");
+                    product.ProductName = Console.ReadLine();
+                    Console.Write("Product new price : ");
+                    product.ProductPrice = Double.Parse(Console.ReadLine());
+                    Console.Write("Product new quantity : ");
+                    product.ProductQuantity = int.Parse(Console.ReadLine());
+                }
+            }
+            Console.WriteLine("Product was not found");
+        }
+
+        public void DeleteProduct(string productName)
+        {
+            foreach(var product in ProductList)
```

Comment:
You can use `SearchForProduct` to return the product and here you can just delete it. Because now you're violating the `DRY` principle, where you are repeating the logic for search for a product in three methods